src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -644,32 +820,42 @@ s32 func_80AF383C(EnRu2* this, PlayState* play) {
     return 0;
 }
 
-void func_80AF3878(EnRu2* this, PlayState* play) {
-    if (func_80AF383C(this, play) && !Play_InCsMode(play)) {
-        this->action = 16;
+/**
+ * Checks if Link is close enough to Ruto and conditionally triggers the encounter cutscene in the Water Temple.
+ */
+void EnRu2_TriggerEncounterInRange(EnRu2* this, PlayState* play) {
+    if (EnRu2_IsPlayerInRange(this, play) && !Play_InCsMode(play)) {
+        this->action = ENRU2_WATER_TEMPLE_ENCOUNTER_BEGIN;
         OnePointCutscene_Init(play, 3130, -99, &this->actor, CAM_ID_MAIN);
     }
 }
 
-void func_80AF38D0(EnRu2* this, PlayState* play) {
-    this->action = 16;
+/**
+ * Triggers the encounter cutscene in the Water Temple, unconditionally. Appears to be unused.
+ */
+void EnRu2_TriggerEncounter(EnRu2* this, PlayState* play) {
+    this->action = ENRU2_WATER_TEMPLE_ENCOUNTER_BEGIN;
     OnePointCutscene_Init(play, 3130, -99, &this->actor, CAM_ID_MAIN);
 }
```

Comment:
This isnt really triggering the encounter right, that already happened above. This is more like SetupBeginEncounter.
Also want to note inconsistency between `ENRU2_WATER_TEMPLE_ENCOUNTER_BEGIN` and `EnRu2_BeginEncounter`

Reply:
No, the difference between this and the other function is that this is unconditional whereas the other one requires Link to be in the same room. Both of them trigger the cutscene where Link meets Ruto, but this one is unused.